hudi-sync/hudi-sync-common/src/main/java/org/apache/hudi/sync/common/AbstractSyncHoodieClient.java

```diff
@@ -178,19 +178,27 @@ public boolean isDropPartition() {
     return false;
   }
 
-  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
   public List<String> getPartitionsWrittenToSince(Option<String> lastCommitTimeSynced) {
-    if (!lastCommitTimeSynced.isPresent()) {
+    return getPartitionsWrittenToSince(lastCommitTimeSynced,false);
+  }
+
+  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+  public List<String> getPartitionsWrittenToSince(Option<String> lastCommitTimeSynced, boolean syncAllPartitions) {
```

Comment:
love to not overload this method with one more path. Can we branch off before in the call stack

Reply:
will do that